Added additional explanations to attributes and supports sections (#57120)
WordPress/gutenberg: docs/getting-started/fundamentals/block-json.md

> ## Data Storage in the Block with `attributes`

Reviewer: ```suggestion
## Using `attributes` to store block data
```

> If the user changes a block, we need a way to persist these changes so they can be loaded and used by the block at a later time. To achieve this, the state of a block is maintained through `attributes` which content is stored in the DB and retrieved when needed. When registering a new block type, the `attributes` property of `block.json` describes the types of data included in the `attributes` object, and how they're stored in the DB so they can be read and passed to the `edit` and `save` functions.

Reviewer: I struggled to understand this section, so here is another suggestion. Also I don't think we should spellout "database" instead of using "DB".

```suggestion
Block [attributes](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#attributes) are settings or data assigned to blocks. They determine various aspects of a block, such as its content, layout, style, and other specific characteristics. If the user changes a block, such as modifying the font size, you need a way to persist these changes. Attributes are the solution. 

When registering a new block type, the `attributes` property of `block.json` describes the custom settings that the block requires and how they're stored in the database. This allows the Editor to parse the attributes correctly and pass them to the block's `Edit` and `save` functions.
```

> Simply put, the [`attributes` property](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#attributes) allows a block to declare "variables" that store data or content for the block.

Reviewer: This can be removed if you like the update above. 

```suggestion
```

	}
},
```

> By default `attributes` are serialized and stored in the block's delimiter, but this [can be configured](https://developer.wordpress.org/news/2023/09/understanding-block-attributes/).

Reviewer: If we aren't talking about the property specifically, I am not sure "attributes" needs to be inline-code.
```suggestion
By default, attributes are serialized and stored in the block's delimiter, but this [can be configured](https://developer.wordpress.org/news/2023/09/understanding-block-attributes/).
```

<!-- /wp:block-development-examples/copyright-date-block-09aac3 -->x
```

> These [`attributes`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#attributes) are passed to the React component `Edit`(to display in the Block Editor) and the `save` function (to return the markup saved to the DB) of the block, and to any server-side render definition for the block (see `block.json`'s `render` property above). 

Reviewer: ```suggestion
These [attributes](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#attributes) are passed to the React component `Edit`(to display in the Block Editor), and the `save` function (to return the markup saved to the database) of the block, and to any server-side render definition for the block. See the `render` property above. 
```

## Enable UI settings panels for the block with `supports`

> A lot of blocks, including core blocks, offer similar customization option, whether that is to change the background color, text color, or to add padding customization options.

Reviewer: ```suggestion
Many blocks, including core blocks, offer similar customization options, whether changing the background color, text color, or adding padding customization options.
```

> To avoid duplicating the same logic over and over in your blocks and to align the behavior of your block with core blocks, you can make use of the different `supports` properties.
> 
> The [`supports`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#supports) property in `block.json` allows a block to declare support for certain features, enabling users to customize specific settings (like colors or margins) from the Settings Sidebar.

Reviewer: ```suggestion
The [`supports`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#supports) property in `block.json` allows a block to declare support for certain features, enabling users to customize specific settings (like colors or margins) from the Settings Sidebar.

Using the available block `supports` allows you to align your block's behavior with core blocks and avoid replicating the same functionality yourself.
```

}
```

> The use of `supports` generates a set of properties that need to be [manually added to the wrapping element of the block](https://developer.wordpress.org/block-editor/getting-started/fundamentals/block-wrapper/) so they're properly stored as part of the block data and taken into account to generate the markup of the block that will be delivered to the front end.

Reviewer: ```suggestion
The use of `supports` generates a set of properties that need to be manually added to the [wrapping element of the block](https://developer.wordpress.org/block-editor/getting-started/fundamentals/block-wrapper/). This ensures they're properly stored as part of the block data and taken into account when generating the markup of the block that will be delivered to the front end.
```
